compiler/rustc_resolve/src/lib.rs

```diff
@@ -2386,7 +2391,40 @@ impl<'a> Resolver<'a> {
                             .next()
                             .map_or(false, |c| c.is_ascii_uppercase())
                         {
-                            (format!("use of undeclared type `{}`", ident), None)
+                            // Add check case for similarly named item in alternative namespace
+                            let mut suggestion = None;
+
+                            if ribs.is_some() {
```

Comment:
```suggestion
                            let suggestion = if ribs.is_some() {
```
(and then return `Some(...)` or `None` inside the `match`)